scheduler-server/scheduling-dsl-compiler/src/main.ts

```diff
@@ -0,0 +1,53 @@
+import Piscina from 'piscina';
+import * as fs from 'fs'
+import * as AST from './libs/scheduler-ast.js';
+
+async function main() {
+
+  var access = fs.createWriteStream('scheduling-dsl-compiler-log.txt', {flags: 'a'});
+
+  // @ts-ignore
+  process.stderr.write = access.write.bind(access);
+
+  console.error("Initial log message")
+
+  process.on('uncaughtException', function(err) {
+    console.error("uncaughtException")
+    console.error((err && err.stack) ? err.stack : err);
+  });
+
+  const workerUrl = new URL('./worker.js', import.meta.url);
+
+  const piscina = new Piscina({
+    filename: workerUrl.href,
+  });
+
+  process.stdin.on("data", async (data: Buffer) => {
+    console.error("data: " + data.toString().trim())
+    try {
+      if (data.toString().trim() === "ping") {
+        // Enable testing the health of the service by sending "ping" and expecting "pong" in return.
+        process.stdout.write("pong\n");
+        return
+      }
+      const { source, filename, generatedCode } = JSON.parse(data.toString()) as { source: string, filename: string, generatedCode: string };
+      const result = await piscina.run({
+        source,
+        filename,
+        generatedCode
+      }) as Promise<{ ast: AST.GoalSpecifier }>;
+      let stringified = JSON.stringify(result);
+      if (stringified === undefined) {
+        throw Error(result + " was not JSON serializable")
+      }
+      console.error("success\n" + stringified + "\n")
+      process.stdout.write("success\n" + stringified + "\n");
+    } catch (error: any) {
+      console.error("error\n" + error.message + '\n')
+      process.stdout.write("error\n" + error.message + '\n');
```

Comment:
Is there a reason why we're duplicating these log messages? Sorry, may be a silly question

Reply:
That's a good question, and pretty key to how this node-java interaction is set up. Java is listening for a response on standard output, so `process.stdout.write` is actually the important functionality here.

`console.error` is just going into the `scheduling-dsl-compiler-log.txt` for debugging purposes.
I think I will get rid of the `console.error`, just because I think it's unnecessarily verbose